### ISSUE TITLE
Notice: Undefined property: Product::$mpn after updating from 1.7.1 ->1.7.7.6

### ISSUE THREAD

**Issue by Henrik41** [PrestaShop/PrestaShop#25609]
After updating I can't add/edit a product:

```
Notice: Undefined property: Product::$mpn

Symfony\Component\Debug\Exception\ContextErrorException:
Notice: Undefined property: Product::$mpn

Uncaught PHP Exception Symfony\Component\Debug\Exception\ContextErrorException: "Notice: Undefined property: Product::$mpn" at /home/gsddev/public_html/src/PrestaShopBundle/Model/Product/AdminModelAdapter.php line 656


  at src/PrestaShopBundle/Model/Product/AdminModelAdapter.php:656
  at PrestaShopBundle\Model\Product\AdminModelAdapter->mapStep6FormData(object(Product))
     (src/PrestaShopBundle/Model/Product/AdminModelAdapter.php:459)
  at PrestaShopBundle\Model\Product\AdminModelAdapter->getFormData(object(Product))
     (src/PrestaShopBundle/Controller/Admin/ProductController.php:667)
  at PrestaShopBundle\Controller\Admin\ProductController->createProductForm(object(Product), object(AdminModelAdapter))
     (src/PrestaShopBundle/Controller/Admin/ProductController.php:457)
  at PrestaShopBundle\Controller\Admin\ProductController->formAction('380', object(Request))
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:151)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:68)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, false)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:200)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request), 1, false)
     (admin1/index.php:82)
```

<img width="1050" alt="Screen Shot 2021-08-15 at 3 35 57 PM" src="https://user-images.githubusercontent.com/81451/129490328-b6bc93e4-5274-4b92-80ce-c3144db6afde.png">

I checked my database and mpn is present where it is suppose too.
My overrides are disabled too.

How do I even debug this?

**Comment by Henrik41** [comment]
Adding to my pain I also have this which tells me that the issue is in mapStep6FormData($product) not reading properly the mpn value of the product from the database:
```

 /home/gsddev/public_html/src/PrestaShopBundle/Model/Product/AdminModelAdapter.php:656 {▼
    › 'upc' => $product->upc,
    › 'mpn' => $product->mpn,
    › 'ean13' => $product->ean13,
  }
  /home/gsddev/public_html/src/PrestaShopBundle/Model/Product/AdminModelAdapter.php:459 {▼
    › $this->formData['step5'] = $this->mapStep5FormData($product);
    › $this->formData['step6'] = $this->mapStep6FormData($product);
    › 
    arguments: {▼
      $product: Product {#1040 ▶}
    }
  }
  /home/gsddev/public_html/src/PrestaShopBundle/Controller/Admin/ProductController.php:667 {▼
    › $formBuilder = $this->createFormBuilder(
    ›     $modelMapper->getFormData($product),
    ›     ['allow_extra_fields' => true]
    arguments: {▼
      $product: Product {#1040 ▼
        +tax_name: null
        +tax_rate: null
        +id_manufacturer: "0"
        +id_supplier: "0"
        +id_category_default: "86"
        +id_shop_default: "1"
        +manufacturer_name: null
        +supplier_name: null
        +name: [▶]
        +description: [▶]
        +description_short: [▶]
        +quantity: -9
        +minimal_quantity: "1"
        +low_stock_threshold: "50"
        +low_stock_alert: "1"
        +available_now: [▶]
        +available_later: [▶]
        +price: "44.000000"
        +specificPrice: 0
        +additional_shipping_cost: "0.000000"
        +wholesale_price: "0.000000"
        +on_sale: "0"
        +online_only: "0"
        +unity: ""
        +unit_price: null
        +unit_price_ratio: "0.000000"
        +ecotax: "0.000000"
        +reference: ""
        +supplier_reference: ""
        +location: ""
        +width: "5.000000"
        +height: "5.000000"
        +depth: "5.000000"
        +weight: "0.500000"
        +ean13: ""
        +isbn: ""
        +upc: ""
        +link_rewrite: [▶]
        +meta_description: [▶]
        +meta_keywords: [▶]
        +meta_title: [▶]
        +quantity_discount: "0"
        +customizable: "0"
        +new: null
        +uploadable_files: "0"
        +text_fields: "0"
        +active: "1"
        +redirect_type: "404"
        +id_type_redirected: "0"
        +available_for_order: "1"
        +available_date: "0000-00-00"
        +show_condition: "0"
        +condition: "new"
        +show_price: "1"
        +indexed: "1"
        +visibility: "both"
        +date_add: "2019-02-25 12:01:48"
        +date_upd: "2021-05-07 11:29:31"
        +tags: false
        +state: "1"
        +base_price: null
        +id_tax_rules_group: "58"
        +id_color_default: 0
        +advanced_stock_management: "0"
        +out_of_stock: 1
        +depends_on_stock: false
        +isFullyLoaded: false
        +cache_is_pack: "0"
        +cache_has_attachments: "1"
        +is_virtual: "0"
        +id_pack_product_attribute: null
        +cache_default_attribute: "0"
        +category: false
        +pack_stock_type: "3"
        +additional_delivery_times: "1"
        +delivery_in_stock: [▶]
        +delivery_out_stock: [▶]
        #webserviceParameters: [▶]
        +id: 256
        #id_lang: null
        #id_shop: 1
        +id_shop_list: []
        #get_shop_from_context: true
        #table: "product"
        #identifier: "id_product"
        #fieldsRequired: [▶]
        #fieldsSize: [▶]
        #fieldsValidate: [▶]
        #fieldsRequiredLang: []
        #fieldsSizeLang: [▶]
        #fieldsValidateLang: [▶]
        #tables: []
        #image_dir: null
        #image_format: "jpg"
        #translator: null
        #def: [▶]
        #update_fields: null
        +force_id: false
        +"image": "https://www.xxxxx.com/511-home_default/eb2fd28tvi.jpg"
      }
    }
  }
  /home/gsddev/public_html/src/PrestaShopBundle/Controller/Admin/ProductController.php:457 {▼
    › 
    › $form = $this->createProductForm($product, $modelMapper);
    › 
    arguments: {▼
      $product: Product {#1040 ▶}
      $modelMapper: AdminModelAdapter {#1009 …}
    }
  }
  /home/gsddev/public_html/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:151 {▶}
  /home/gsddev/public_html/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:68 {▶}
  /home/gsddev/public_html/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:200 {▶}
  /home/gsddev/public_html/admin1/index.php:82 {▼
    › require_once __DIR__.'/../autoload.php';
    › $response = $kernel->handle($request, HttpKernelInterface::MASTER_REQUEST, false);
    › $response->send();
    arguments: {▶}
  }
}
```

As you can see mpn is not included in product#1040, Therefore the $product is not reading mpn field for whatever reason I can't explain. I have no clue how it even gets $product. Since Prestashop is a spaghetti that only aliens can read.

I removed the mpn field on step 6 and  the error is gone. This is only a temporary solution until I find out why $product does not contain the $mpn field. I suspect that an old file from 1.7.1 (still present after upgrade) is interacting with the new Adapter and doesn't read $product properly.

**Comment by hibatallahAouadni** [comment]
Hello @Henrik41 

We use GitHub issues only to discuss about bugs and new features in the PrestaShop project. If you need help with your shop, please consider one of our [support plans](https://www.prestashop.com/support).

Alternatively, you can also ask for help in the [community forums](https://www.prestashop.com/forums/) or in the [public Slack channel](https://github.com/PrestaShop/open-source/tree/master/slack).

Thank you
